File: Csharp/1704.cs
public class Solution
{
    public bool HalvesAreAlike(string s)
    {
        //splitting a string in 2
        s = s.ToLower();
        int half = s.Length / 2;

        char[] firstHalf = s.Take(half).ToArray();
        char[] secondHalf = s.Skip(half).ToArray();

        //count how mahy vowles each contains
        int a = countVowles(firstHalf);
        int b = countVowles(secondHalf);


        //if they are equal retgurn true otherwise false
        if (a == b) return true;
        else return false;

    }

    public int countVowles(char[] message)
    {
        int a = 0;
        foreach (var letter in message)
        {
            if (letter == 'a' || letter == 'e' || letter == 'i' || letter == 'o' || letter == 'u') a++;
        }
        return a;
    }
}

File: JS/1704.js
/**
 * @param {string} s
 * @return {boolean}
 */
var halvesAreAlike = function(s) {
    let sLower = s.toLowerCase();
    let half = sLower.length /2;
    let firstHalf = [...sLower];
    let secondHalf = firstHalf.splice(half,sLower.length);
    let count = countVowels(firstHalf)
    let count2 = countVowels(secondHalf);

    if(count === count2){
        return true;
    } else {
        return false;
    }
};

var countVowels = function(word){
    let count = 0;
    for(let i = 0; i < word.length; i++){
        if(word[i] === 'a' || word[i] === 'e' || word[i] === 'i' || word[i] === 'o' || word[i] === 'u'){
            count++;
        }
    }
    return count;
}